pylint/config/option_manager_mixin.py

```diff
@@ -305,23 +304,36 @@ def _parse_toml(
         except KeyError:
             return
         for section, values in sections_values.items():
+            section_name = section.upper()
             # TOML has rich types, convert values to
             # strings as ConfigParser expects.
+            if not isinstance(values, dict):
+                # This class is a mixin the add_message come from the pylinter
+                self.add_message(  # type: ignore
+                    "bad-configuration-option-value", line=0, args=(section, values)
```

Comment:
I'm probably overlooking something really obvious, but how does this check whether `load-plugins = []` should be in the `tools.pylint` section (as one of the tests does)?

Reply:
There is nothing directly in the ``tools.pylint`` except header (master, message controls, etc), so at top level, if it's not a dict it can't be right. We know nothing about what to expect in the configuration except that.

Reply:
Perhaps `bad-configuration-section` would be better then? For me `option-value` implies there is something wrong with the type of the option value (but that might be just me).

I think such a check (option value types) would be helpful anyway, if we don't already check it, so reserving `bad-configuration-option-value` for that checker would be better. 
Putting stuff in the wrong section and giving stuff the incorrect type are different mistakes and require different messages. For the the latter we could (for example) also include the expected type in the message.

Reply:
I agree, I'm going to change the name and upgrade the tests.

Reply:
22ed9232